sematic/ui/packages/common/src/pages/RunSearch/SearchFilters.tsx

```diff
@@ -80,8 +82,20 @@ const SearchFilters = (props: SearchFiltersProps) => {
         onFiltersChanged({...allFilters.current});
     }, [onFiltersChanged]);
 
+    const [submitedDefaults, setSubmitedDefaults] = useState(false);
+    useEffect(() => {
+        if(submitedDefaults) {
+            return;
+        }
+        if(!!defaultSearchString) {
+            allFilters.current[FilterType.SEARCH] = [defaultSearchString];
+        }
+        applyFilters();
+        setSubmitedDefaults(true);
+    }, [submitedDefaults, setSubmitedDefaults, applyFilters, allFilters, defaultSearchString]);
+
     return <>
-        <SearchTextSection ref={searchTextRef} onSearchChanged={onSearchTextChanged} />
+        <SearchTextSection ref={searchTextRef} onSearchChanged={onSearchTextChanged} defaultSearchString={defaultSearchString} />
```

Comment:
Yes, this works. The more natural way is to have the `<SearchTextSection>` detects the deeplink information and hydrate its state within itself, then report back to parent (`< earchFilters>`) using `onSearchChanged` in its useEffect. Then in parent's useEffect, we simply call `applyFilter()`. Because the parent's useEffect is triggered after child's useEffect, it should work.

Reply:
@augray FYI, I implemented it in https://github.com/sematic-ai/sematic/pull/943